### DISCUSSION TITLE
Should `intel-mkl` fetch from NuGet?

### DISCUSSION THREAD

**Discussion by ras0219-msft** [microsoft/vcpkg#21478] [Ideas]
It looks like the Intel MKL libraries are available via NuGet: https://www.intel.com/content/www/us/en/develop/documentation/installation-guide-for-intel-oneapi-toolkits-windows/top/installation/install-using-package-managers/nuget.html

We could use this to acquire them without forcing the user to jump through all the hoops of the full SDK installation. However, they are _big_; the mkl-core.lib static library for x64-win is 600 MB.

Thoughts? I think this is probably a positive change, but perhaps there're other factors I'm unaware of (like the library won't load without also having admin-installed some crazy drivers and so users need to get the SDK anyway).

+@Neumann-A  @cenit for scientific computing advice

Example: https://www.nuget.org/packages/intelmkl.redist.win-x86/

**Comment by cenit** [comment]
when cudnn was setup in a similar way, biggest complains came from users that had the sdk installed, because for them the library managed through vcpkg meant only unnecessary duplication of very big library in each project, while it was shared for  installed sdk. Accepting both in portfile violated binary cache fidelity as set up now. Apart from that, thumbs up for me
